pkg/source/source.go

```diff
@@ -221,6 +222,28 @@ func (cs *Channel) syncLoop() {
 	}
 }
 
+// Informer is used to provide a source of events originating inside the cluster from Watches (e.g. Pod Create)
+type Informer struct {
+	// Informer is the generated client-go Informer
+	Informer toolscache.SharedIndexInformer
+}
+
+var _ Source = &Informer{}
+
+// Start is internal and should be called only by the Controller to register an EventHandler with the Informer
+// to enqueue reconcile.Requests.
+func (ks *Informer) Start(handler handler.EventHandler, queue workqueue.RateLimitingInterface,
+	prct ...predicate.Predicate) error {
+
+	// Informer should have been specified by the user.
+	if ks.Informer == nil {
+		return fmt.Errorf("must specify Informer.Informer")
+	}
+
+	ks.Informer.AddEventHandler(internal.EventHandler{Queue: queue, EventHandler: handler, Predicates: prct})
+	return nil
+}
+
```

Comment:
Thinking more about the code in the manager's `main` fn, I am wondering if we need to add the concept of `Source` being ready (in this case, informed being sync'ed) from the perspective of wiring everything. Will have to look at the code to reason more about it.